hail/python/hailtop/aiogoogle/client/compute_client.py

```diff
@@ -77,13 +82,50 @@ async def list(self, path: str, *, params: MutableMapping[str, Any] = None, **kw
         return PagedIterator(self, path, params, kwargs)
 
     async def create_disk(self, path: str, *, params: MutableMapping[str, Any] = None, **kwargs):
-        return await request_with_wait_for_done(self.post, path, params, **kwargs)
+        return await self._request_with_zonal_operations_response(self.post, path, params, **kwargs)
 
     async def attach_disk(self, path: str, *, params: MutableMapping[str, Any] = None, **kwargs):
-        return await request_with_wait_for_done(self.post, path, params, **kwargs)
+        return await self._request_with_zonal_operations_response(self.post, path, params, **kwargs)
 
     async def detach_disk(self, path: str, *, params: MutableMapping[str, Any] = None, **kwargs):
-        return await request_with_wait_for_done(self.post, path, params, **kwargs)
+        return await self._request_with_zonal_operations_response(self.post, path, params, **kwargs)
 
     async def delete_disk(self, path: str, *, params: MutableMapping[str, Any] = None, **kwargs):
-        return await request_with_wait_for_done(self.delete, path, params, **kwargs)
+        return await self.delete(path, params=params, **kwargs)
+
+    async def _request_with_zonal_operations_response(self, request_f, path, params: MutableMapping[str, Any] = None, **kwargs):
+        assert 'params' not in kwargs
+
+        async def request_and_wait():
+            local_params = copy.deepcopy(params)
+
+            if local_params is None:
+                local_params = {}
+
+            request_uuid = str(uuid.uuid4())
+            if 'requestId' not in local_params:
+                local_params['requestId'] = request_uuid
+
+            resp = await request_f(path, params=local_params, **kwargs)
+            operation_id = resp['id']
+            zone = resp['zone'].rsplit('/', 1)[1]
+
+            while True:
+                result = await self.post(f'/zones/{zone}/operations/{operation_id}/wait')
+                if result['status'] == 'DONE':
+                    http_error_status_code = result.get('httpErrorStatusCode')
+                    http_error_message = result.get('httpErrorMessage')
+                    errors = None
+
+                    error = result.get('error')
+                    if error:
+                        errors = error['errors']
+                        errors = [GCPError(e['code'], e['message']) for e in errors]
+
+                    if http_error_status_code is not None:
+                        raise GCPOperationError(http_error_status_code, http_error_message, errors, result)
+                    return result
+
+                await asyncio.sleep(1)
```

Comment:
Hmm. I don't like a hardcoded 1 here. I feel like we should use exponential back off like we do elsewhere. `wait` returns if:
1. The operation finished.
2. The server doesn't want to hold open the operation (probably because its overloaded).
3. We're approaching the 2 minute deadline.

In the first case, we won't reach the sleep. In the second case, we should definitely exponentially back off. In the third case, we don't need to sleep at all. I think we can satisfy all cases by exponentially backing off with a maximum wait much smaller than 2 minutes. Maybe 15 seconds? What do you think?